app/domain/authentication/authn_jwt/restriction_validation/validate_restrictions_one_to_one.rb

```diff
@@ -23,12 +25,11 @@ def valid_restriction?(restriction)
             raise Errors::Authentication::ResourceRestrictions::EmptyAnnotationGiven, annotation_name
           end
 
-          unless @decoded_token.key?(claim_name)
-            raise Errors::Authentication::AuthnJwt::JwtTokenClaimIsMissing,
-                  claim_name_for_error(annotation_name, claim_name)
-          end
+          claim_value = @decoded_token.dig(*parsed_claim_path(claim_name))
```

Comment:
Move right side to function called claim_value

Reply:
why?

Reply:
To seperate responsibilities between function. One wrapping parsed_claim_path and one doing the compare. Do what you think its not critical

Reply:
It's replace one function `dig` by other.
The class is stateless from single restriction perspective so memorization is impossible.

Reply:
Without memoriaztion jsut
```
def claim_value(claim_name)
   @decoded_token.dig(*parsed_claim_path(claim_name))
end 
```

So the comprare
restriction_value == claim_value

will be the only call for the function.

Reply:
there are multiple usages by claim value in the main function
I don't want to dig for value each time it needed
It's enough to dig once.